src/Game/AI.cpp

```diff
@@ -423,4 +423,142 @@ namespace ai{
+            }
+        }else if(activeEntityId == communication::messages::types::EntityId::LEFT_TROLL ||
+                 activeEntityId == communication::messages::types::EntityId::RIGHT_TROLL){
+            if(isTrollUseful(mySide, env)){
+                return communication::messages::request::DeltaRequest{communication::messages::types::DeltaType::TROLL_ROAR, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
+                                                                      std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt};
+            }else{
+                return communication::messages::request::DeltaRequest{communication::messages::types::DeltaType::SKIP, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
+                                                                      std::nullopt, activeEntityId, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt};
+            }
+        }else if(activeEntityId == communication::messages::types::EntityId::LEFT_GOBLIN ||
+                 activeEntityId == communication::messages::types::EntityId::RIGHT_GOBLIN){
+            passiveEntityId = getGoblinTarget(mySide, env);
+            if(passiveEntityId.has_value()){
+                return communication::messages::request::DeltaRequest{communication::messages::types::DeltaType::GOBLIN_SHOCK, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
+                                                                      std::nullopt, std::nullopt, passiveEntityId, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt};
+            }else{
+                return communication::messages::request::DeltaRequest{communication::messages::types::DeltaType::SKIP, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
+                                                                      std::nullopt, activeEntityId, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt};
+            }
+        }else{
+            return communication::messages::request::DeltaRequest{communication::messages::types::DeltaType::SKIP, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
+                                                                  std::nullopt, activeEntityId, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt};
+        }
+    }
+
+    bool isNifflerUseful(const gameModel::TeamSide &mySide, const std::shared_ptr<gameModel::Environment> &env){
+        if (mySide == env->team1->side) {
+            return getDistance(env->team2->seeker->position, env->snitch->position) <= 2;
+        }else{
+            return getDistance(env->team1->seeker->position, env->snitch->position) <= 2;
+        }
+    }
+
+    bool isTrollUseful(const gameModel::TeamSide &mySide, const std::shared_ptr<gameModel::Environment> &env) {
+        if (mySide == env->team1->side) {
```

Comment:
Hol dir einfach den Spieler auf der Position des Balls. Wenn da einer drauf steht, prüfe, ob er in deinem Team ist.